backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/workpackage/impl/AbstractOrderLineLog.java

```diff
@@ -100,53 +86,61 @@ public SOLineForSOLogHandler(@NonNull final ModularContractService modularContra
 				.orElse(null);
 
+		final String productName = productBL.getProductValueAndName(productId);
+		final SOTrx soTrx = SOTrx.ofBoolean(order.isSOTrx());
+		final AdMessageKey msgToUse = soTrx.isSales() ? MSG_SO_ON_COMPLETE_DESCRIPTION : MSG_PO_ON_COMPLETE_DESCRIPTION;
+		final String description = msgBL.getBaseLanguageMsg(msgToUse, productName, quantity.abs().toString());
+
 		return ExplainedOptional.of(LogEntryCreateRequest.builder()
-											.contractId(createLogRequest.getContractId())
-											.productId(productId)
-											.productName(createLogRequest.getProductName())
-											.referencedRecord(TableRecordReference.of(I_C_OrderLine.Table_Name, orderLine.getC_OrderLine_ID()))
-											.producerBPartnerId(BPartnerId.ofRepoId(order.getC_BPartner_ID()))
-											.invoicingBPartnerId(BPartnerId.ofRepoId(order.getBill_BPartner_ID()))
-											.collectionPointBPartnerId(BPartnerId.ofRepoId(warehouseRecord.getC_BPartner_ID()))
-											.warehouseId(warehouseId)
-											.documentType(getLogEntryDocumentType())
-											.contractType(LogEntryContractType.MODULAR_CONTRACT)
-											.soTrx(SOTrx.SALES)
-											.processed(false)
-											.quantity(quantity)
-											.amount(amount)
-											.transactionDate(transactionDate)
-											.year(yearAndCalendarId.yearId())
-											.description(description)
-											.modularContractTypeId(createLogRequest.getTypeId())
-											.configModuleId(createLogRequest.getConfigId().getModularContractModuleId())
-											.priceActual(orderLineBL.getPriceActual(orderLine))
-											.invoicingGroupId(invoicingGroupId)
-											.build());
+				.contractId(createLogRequest.getContractId())
+				.productId(productId)
+				.productName(createLogRequest.getProductName())
+				.referencedRecord(recordRef)
+				.producerBPartnerId(BPartnerId.ofRepoId(order.getC_BPartner_ID()))
+				.invoicingBPartnerId(BPartnerId.ofRepoId(order.getBill_BPartner_ID()))
+				.collectionPointBPartnerId(BPartnerId.ofRepoId(order.getC_BPartner_ID()))
+				.warehouseId(WarehouseId.ofRepoId(order.getM_Warehouse_ID()))
+				.documentType(getLogEntryDocumentType())
+				.contractType(LogEntryContractType.MODULAR_CONTRACT)
+				.soTrx(soTrx)
+				.processed(false)
+				.quantity(quantity)
+				.amount(amount)
+				.transactionDate(transactionDate)
+				.year(yearAndCalendarId.yearId())
+				.description(description)
+				.modularContractTypeId(createLogRequest.getTypeId())
+				.configModuleId(createLogRequest.getConfigId().getModularContractModuleId())
+				.priceActual(orderLineBL.getPriceActual(orderLine))
+				.invoicingGroupId(invoicingGroupId)
+				.isBillable(false)
+				.build());
 	}
 
 	@Override
 	public @NonNull ExplainedOptional<LogEntryReverseRequest> createLogEntryReverseRequest(@NonNull final CreateLogRequest createLogRequest)
 	{
-		throw new UnsupportedOperationException();
+		return ExplainedOptional.of(LogEntryReverseRequest.builder()
```

Comment:
can we reverse orders now? If so, what happens to the contracts on order reversal?

Reply:
No, this was pulled up from purchaseOrderLine implementation, but atm is never reached. We have `de.metas.contracts.modular.computing.ComputingMethodService#validateAction`

Reply:
Then let's kick it, to not confuse the reader into thinking the reversal is possible. Maybe use ``		throw new AdempiereException(MSG_ERROR_DOC_ACTION_UNSUPPORTED);``

Reply:
done